weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/Charts.tsx

```diff
@@ -0,0 +1,263 @@
+import {quantile} from 'd3-array';
+import _ from 'lodash';
+import moment from 'moment';
+import * as Plotly from 'plotly.js';
+import React, {useEffect, useMemo, useRef} from 'react';
+
+import {
+  BLUE_500,
+  GREEN_500,
+  MOON_200,
+  MOON_300,
+  MOON_500,
+  MOON_750,
+  TEAL_400,
+} from '../../../../../../common/css/color.styles';
+
+type ChartDataRequests = {
+  started_at: string;
+};
+
+type ChartDataErrors = {
+  started_at: string;
+  isError: boolean;
+};
+
+type ChartDataLatency = {
+  started_at: string;
+  latency: number;
+};
+
+const CHART_TITLE_STYLE = {
+  font: {
+    color: MOON_750,
+    family: '"Source Sans Pro", sans-serif',
+    size: 16,
+    weight: 700,
+  },
+  pad: {
+    l: 16,
+  },
+  x: 0,
+  xanchor: 'left' as const,
+};
+
+const CHART_MARGIN_STYLE = {
+  l: 50,
+  r: 30,
+  b: 50,
+  t: 50,
+  pad: 0,
+};
+
+const X_AXIS_STYLE: Partial<Plotly.LayoutAxis> = {
+  type: 'date' as const,
+  automargin: true,
+  showgrid: false,
+  linecolor: MOON_300,
+  showspikes: true,
+  spikemode: 'across',
+  spikethickness: 1,
+  spikecolor: MOON_300,
+  tickfont: {color: MOON_500},
+};
+
+const Y_AXIS_STYLE: Partial<Plotly.LayoutAxis> = {
+  automargin: true,
+  griddash: 'dot',
+  showgrid: true,
+  gridcolor: MOON_300,
+  linecolor: MOON_300,
+  showspikes: false,
+  tickfont: {color: MOON_500},
+  zeroline: false,
+};
+
+export const LatencyPlotlyChart: React.FC<{
+  height: number;
+  chartData: ChartDataLatency[];
+  binSizeMinutes?: number;
+}> = ({height, chartData, binSizeMinutes = 60}) => {
+  const divRef = useRef<HTMLDivElement>(null);
+
+  const plotlyData: Plotly.Data[] = useMemo(() => {
+    const groupedData = _(chartData)
+      .groupBy(d => {
+        const date = moment(d.started_at);
+        const roundedMinutes =
+          Math.floor(date.minutes() / binSizeMinutes) * binSizeMinutes;
```

Comment:
Grouping by the hour, we can change this if there's a preference

Reply:
I think the binning logic really wants to be:
1. Select all the data given the filters and count
2. Calculate the delta between the first and last result ("totalTimeRange")
3. Divide "TotalTimeRange" by a configurable, fixed number (say 30), this will give you "BinSize" - this should be the same bin size across all charts.
4. Allocate each call into the corresponding bin.

Fixing the bin size is not a viable since it doesn't adapt to the traffic volume.

Reply:
I think I see what you're saying -- grouping by the hour won't always work since our data may only be over one hour or less for example (or too many hours). We can calculate the bin size based on a fixed number